File: src/main/java/no/nav/pto/veilarbportefolje/hendelsesfilter/domain/Hendelse.java
package no.nav.pto.veilarbportefolje.hendelsesfilter.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public record Hendelse(
        String personID,
        String avsender,
        Kategori kategori,
        Operasjon operasjon,
        @JsonProperty(value = "hendelse")
        HendelseInnhold hendelseInnhold
) {
    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
    public Hendelse {
    }
}

File: src/main/java/no/nav/pto/veilarbportefolje/hendelsesfilter/domain/Hendelse.kt
package no.nav.pto.veilarbportefolje.hendelsesfilter.domain

import com.fasterxml.jackson.annotation.JsonProperty

data class Hendelse(
    val personID: String,
    val avsender: String,
    val kategori: Kategori,
    val operasjon: Operasjon,
    @JsonProperty(value = "hendelse")
    val hendelseInnhold: HendelseInnhold
)